examples/cli-jobs/upload-local-data/confidential_io.py

```diff
@@ -0,0 +1,215 @@
+import os
+import io
+from azure.keyvault.keys import KeyClient
+from azure.keyvault.keys.crypto import CryptographyClient, EncryptionAlgorithm
+from azure.identity import DefaultAzureCredential, ManagedIdentityCredential
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import padding
+import logging
+import warnings
+from distutils.util import strtobool
+
+_MANAGED_IDENTITY = None
+_KEYVAULT_URL = None
+_RSA_KEY_NAME = None
+_RSA_CRYPTO_CLIENT = None
+
+# WARNING !!!
+_CONFIDENTIALITY_DISABLED = bool(
+    strtobool(os.environ.get("CONFIDENTIALITY_DISABLE", "False"))
+)
+
+if _CONFIDENTIALITY_DISABLED:
+    warnings.warn(
+        "confidentiality has been intentionally disabled using CONFIDENTIALITY_DISABLE=True, all outputs will be left in clear."
+    )
+
+
+def config_global_rsa_key(
+    keyvault_url=None, rsa_key_name=None, managed_identity=None, lazy_init=True
+):
+    global _KEYVAULT_URL, _RSA_KEY_NAME, _MANAGED_IDENTITY
+    _KEYVAULT_URL = (
+        _KEYVAULT_URL
+        or keyvault_url
+        or os.environ.get("CONFIDENTIALITY_KEYVAULT", None)
+    )
+    _RSA_KEY_NAME = (
+        _RSA_KEY_NAME
+        or rsa_key_name
+        or os.environ.get("CONFIDENTIALITY_KEY_NAME", None)
+    )
+    _MANAGED_IDENTITY = (
+        _MANAGED_IDENTITY
+        or managed_identity
+        or os.environ.get("DEFAULT_IDENTITY_CLIENT_ID", None)
+    )
+
+    if not lazy_init:
```

Comment:
Is this used anywhere, cannot we just omit the `lazy_init`?

Reply:
not in the examples, by default I'm initializing the RSA client at the first call of config_global_rsa_key(), the intent was to test connection with the keyvault, but I'm not even sure the library does that, I'll remove this option.